src/OrchardCore/OrchardCore.Navigation.Core/NavigationManager.cs

```diff
@@ -203,8 +208,13 @@ private string GetUrl(string menuItemUrl, RouteValueDictionary routeValueDiction
                 menuItemUrl = menuItemUrl.Substring(2);
             }
 
+            if (menuItemUrl.StartsWith(_adminOptions.AdminUrlPrefix, StringComparison.OrdinalIgnoreCase))
+            {
+                menuItemUrl = menuItemUrl.Remove(0, _adminOptions.AdminUrlPrefix.Length); //backward compatibility
```

Comment:
In case the 1st segment starts with the prefix but not only until the first `/`, we need to check if it starts with `AdminUrlPrefix` + `/`. Then here in place of removing a segment to be re-added below, we can just return `...PathBase.Add('/' + menuItemUrl).Value` as before. Notice that currently here it seems that we keep the first `/` which would end up with a double slashes (line 217).

---

One suggestion would be to use a pattern convention, if the menuItemUrl contains for example `{admin}`, just replace this token with the current admin prefix option, otherwise if no token it works exactly as before.

Reply:
@jtkech this is handled before this condition itself

https://github.com/OrchardCMS/OrchardCore/pull/12807/files/e973fdf6adae4df5fcd88dd04bb33f69e1f5bd36#diff-09302c08752ef136263675f9d9bc818bbd2f2a5fcaa00eb7a684c738ef82dfd2L195

Reply:
I'm talking about the first `/` after the admin prefix, to differentiate for example `currentprefix123/other/segment` from `currentprefix/other/segment`, so we should ckeck if it starts with `currentprefix/` with a trailing `/`.

Then line 213 it seems that we don't remove the `/` after this `currentprefix` and another one is added line 217.

Then line 213 I suggested to just return `...PathBase.Add('/' + menuItemUrl).Value` so that no need to remove a segment and re-add it line 217.

Finally I suggested another approach by using a token in the url as a convention, for example `{admin}`, that would be easier to replace if a menuItemUrl contains this token.

    {admin}/other/segments    =>    currentprefix/other/segments.
    admin/other/segments    =>    admin/other/segments // nothing replaced if no token.

Reply:
Got it, Let me try and update here soon. Thanks for the suggestions.

Reply:
@jtkech I have returned the menu item URL directly instead of removing and adding again.